Apps/Examples/Examples/All Examples/Custom2DPuckExample.swift

```diff
@@ -4,27 +4,125 @@ import MapboxMaps
 @objc(Custom2DPuckExample)
 public class Custom2DPuckExample: UIViewController, ExampleProtocol {
 
-    internal let toggleAccuracyRadiusButton: UIButton = UIButton(frame: .zero)
     internal var mapView: MapView!
-    internal var showsAccuracyRing: Bool = false {
+    internal var puckConfiguration = Puck2DConfiguration.makeDefault(showBearing: true)
+    private var showsPuck: PuckVisibility = .isVisible {
         didSet {
-            syncPuckAndButton()
+            mapView.location.options.puckType = showsPuck == .isVisible ? .puck2D(puckConfiguration) : .none
+        }
+    }
+
+    private var puckImage: PuckImage = .blueDot {
+        didSet {
+            puckConfiguration.topImage = puckImage == .blueDot ? .none : UIImage(named: "star")
+            // Only add puck with new image to the map if it is meant to be visible
+            if showsPuck == .isVisible {
+                mapView.location.options.puckType = .puck2D(puckConfiguration)
+            }
+        }
+    }
+
+    private var showsBearing: PuckBearingVisibility = .isVisible {
+        didSet {
+            // It's necessary to create a new default when switching bearing visibility
+            var newPuck = showsBearing == .isVisible ? Puck2DConfiguration.makeDefault(showBearing: true) : Puck2DConfiguration.makeDefault(showBearing: false)
+            newPuck.topImage = puckConfiguration.topImage
+            newPuck.showsAccuracyRing = puckConfiguration.showsAccuracyRing
+            if showsPuck == .isVisible {
+                mapView.location.options.puckType = .puck2D(newPuck)
+            }
+            puckConfiguration = newPuck
+        }
+    }
+
+    private var showsAccuracyRing: PuckAccuracyRingVisibility = .isHidden {
+        didSet {
+            puckConfiguration.showsAccuracyRing = showsAccuracyRing == .isVisible ? true : false
+            if showsPuck == .isVisible {
+                mapView.location.options.puckType = .puck2D(puckConfiguration)
+            }
+        }
+    }
+
+    private var bearingSource: PuckBearingSource = .course {
+        didSet {
+            mapView.location.options.puckBearingSource = bearingSource
+        }
+    }
+
+    private var style: Style = .dark {
+        didSet {
+            mapView.mapboxMap.style.uri = style == .light ? StyleURI.light : StyleURI.dark
+        }
+    }
+
+    private var projection: StyleProjectionName = .mercator {
+        didSet {
+            do {
+                try mapView.mapboxMap.style.setProjection(StyleProjection(name: projection))
+            } catch {
+                print(error)
+            }
+        }
+    }
+
+    private enum PuckVisibility {
```

Comment:
is there a reason why are using a custom enum here instead of a boolean?

Reply:
I fully support custom enums over boolean types due to better readability in code.